src/Socket.bs

```diff
@@ -0,0 +1,84 @@
+' /**
+'  * @module Socket
+'  */
+namespace Socket
+class SocketConnection
+
+public socket = invalid
+public connection = invalid
+
+public function CreateSocketAndWaitConnection(port) as void
+
+    messagePort = CreateObject("roMessagePort")
+    m.socket = CreateObject("roStreamSocket")
+    m.socket.setMessagePort(messagePort)
+    addr = CreateObject("roSocketAddress")
+    addr.setPort(port)
+    m.socket.setAddress(addr)
+    m.socket.notifyReadable(true)
+    x = m.socket.listen(1)
+    if not m.socket.eOK()
+        print "Error creating listen socket" 
+    end if
+
+    print "Waiting for socket connection ..." 
+
+    while True
+        msg = wait(0, messagePort)
+        if type(msg) = "roSocketEvent"
+            changedID = msg.getSocketID()
+            if changedID = m.socket.getID() and m.socket.isReadable()
+                newConnection = m.socket.accept()
+                if newConnection = invalid
+                    print "New connection failed"
+                else
+                    print "Accepted new connection: " newConnection.getID() 
+                    m.connection = newConnection                    
+                    return
+                end if
+            else
+                if closed or not newConnection.eOK()
+                    print "Closing connection " changedID
+                    newConnection.close()
+                end if
+            end if
+        end if
+    end while
+
+end function
+
+
+public function sendResults(msg) as void
+
+    socketResultMsg = {
+        success: false, 
+        testCount: 0
+        failedTests: 0
+        failedTestsNames: []
+        failedFiles: []
+        failureMessages: []
+    }
+
+    if msg.testRunHasFailures = false
+        socketResultMsg.success = true
+    end if
+
+    socketresultmsg.testCount = msg.total
+    socketresultmsg.failedTests = msg.fail
+
+    for each suite in msg.suites
+        for each test in suite.tests
```

Comment:
Trouble is metatestcase has children that are functions, so we can't call `formatJSON` on it... >BRIGHTSCRIPT: ERROR: FormatJSON: Value type not supported: Function

Reply:
We should add a function to metaTestSuite, like `getJson`, which will return exactly what we want. we'll likely want the same function on `TestCase` too, as it will cascade

Reply:
OK, will have look at doing that.